app/pages/sections/order-items/catalogue-solutions/select/routes.js

```diff
@@ -80,6 +80,14 @@ export const catalogueSolutionsSelectRoutes = (authProvider, addContext, session
         catalogueItemsKey: 'solutions',
       });
 
+      const orderItems = sessionManager.getFromSession({ req, key: sessionKeys.orderItems });
+      const alreadySelectedItem = orderItems
+        .filter((orderItem) => orderItem.catalogueItemId === selectedItem.catalogueItemId);
+      if (alreadySelectedItem.length > 0 && alreadySelectedItem[0].catalogueItemId) {
+        return res.redirect(
+          `${config.baseUrl}/organisation/${orderId}/catalogue-solutions/${alreadySelectedItem[0].catalogueItemId}`,
+        );
+      }
```

Comment:
What happens if we don't have the order items in session, such as if the user has created an order without completing it and comes back the next day? I think the catalogue solutions selection page will still be displayed in that case – is that correct? 

Therefore, we'll need to call the API (to get a list of the order items) if we don't have any order items in session won't we? We have a `getFromSessionOrApi` helper function, you can see some examples of its use in `helpers/routes/` (such as `getOrderDescription`). Could we use something similar here?

Reply:
thanks Chris , will check that scenario and getFromSessionOrApi helper

Reply:
Thanks for helping resolve my misunderstanding @nehajain008!